migrations/20190110193710_steps_table.js

```diff
@@ -0,0 +1,18 @@
+
+exports.up = function(knex, Promise) {
+    return knex.schema.createTable("steps", steps=>{
+        steps.increments();
+        steps.string('name',255).notNullable();
+        steps.string('step').notNullable();
+        steps
+        .integer("instruction_id")
+        .unsigned()
+        .references("id")
+        .inTable("instructions")
+        .notNullable();
```

Comment:
There's a surprising amount of repetiton with the creation of these tables. One of the most common and length patterns I see if your foreign keys. What if we broke that out and used a helper function? For example, have this stored in a separate file:
```javascript
module.exports = {
  foreign: (table, ref) => table
    .integer(`${ref}_id`)
    .unsigned()
    .references('id')
    .on(`${ref}s`)
    .notNullable()
}; 
```
and then the usage in this migration would be:
```javascript
const { foreign } = require('./path_to_helper')
exports.up = function(knex, Promise) {
    return knex.schema.createTable("steps", steps=>{
        steps.increments();
        steps.string('name',255).notNullable();
        steps.string('step').notNullable();
        foreign(steps, 'instruction');
    })
};
```